### PR TITLE
Task#116 | Maze builder info(difficult)

### PR DESCRIPTION
**Pull request by Kadabras**

### CODE REVIEW

**Review by JuliaGorbach21 — COMMENTED** [review]
Reviewed

File: Net12/WebMaze/Models/MazeDifficultProfileViewModel.cs [hunk 1]
```diff
@@ -0,0 +1,21 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace WebMaze.Models
+{
+    public class MazeDifficultProfileViewModel
+    {
+        public long Id { get; set; }
```
Comment (GORYNTCHdotNET) [inline Net12/WebMaze/Models/MazeDifficultProfileViewModel.cs:10]: Why do u use the "Id" inside the ViewModel?

**Review by GORYNTCHdotNET — APPROVED** [review]

**Review by GORYNTCHdotNET — APPROVED** [review]

File: Net12/WebMaze/Controllers/MazeController.cs [hunk 2]
```diff
@@ -1,33 +1,67 @@
...
-            var model = new CellInfoViewModel();
+            return View();
+        }
+
+        [HttpPost]
+        public IActionResult AddMazeDifficult(MazeDifficultProfileViewModel mazeDifficultProfileViewModel)
+        {
+            var dbMazeDifficult = _mapper.Map<MazeDifficultProfile>(mazeDifficultProfileViewModel);
+            dbMazeDifficult.IsActive = true;
+            dbMazeDifficult.Creater = _mazeDifficultRepository.GetRandomUser();
```
Comment (JuliaGorbach21) [inline Net12/WebMaze/Controllers/MazeController.cs:44]: You can use the UserRepository method "GetRandomUser" here. Add the parametr of this type and write it in the constructor.

File: Net12/WebMaze/EfStuff/Repositories/MazeDifficultRepository.cs [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+using WebMaze.EfStuff.DbModel;
+
+namespace WebMaze.EfStuff.Repositories
+{
+    public class MazeDifficultRepository : BaseRepository<MazeDifficultProfile>
+    {
+        public MazeDifficultRepository(WebContext webContext) : base(webContext)
+        {
+        }
+        public User GetRandomUser()
```
Comment (JuliaGorbach21) [inline Net12/WebMaze/EfStuff/Repositories/MazeDifficultRepository.cs:14]: If you use a UserRepository method you can remove this method from here.

File: Net12/WebMaze/Views/Shared/_Layout.cshtml [hunk 4]
```diff
@@ -64,6 +64,10 @@
                     <li class="nav-item">
                         <a class="nav-link" href="/CellInfo/AddCell">Add Cell</a>
                     </li>
+                    <li class="nav-item">
+                        <a class="nav-link" href="/Maze/Index">Maze</a>
```
Comment (JuliaGorbach21) [inline Net12/WebMaze/Views/Shared/_Layout.cshtml:68]: Mae be <a class="nav-link" href="/Maze/ManageMazeDifficult">ManageMazeDifficult</a>

**Review by JuliaGorbach21 — COMMENTED** [review]
Reviewed

File: Net12/WebMaze/Views/Shared/_Layout.cshtml (hunk 4 above)
Comment (Kadabras) [reply Net12/WebMaze/Views/Shared/_Layout.cshtml:68]: no

**Pull request merged**